pkg/block/azure/adapter.go

```diff
@@ -27,25 +31,36 @@ const (
 	idSuffix   = "_id"
 	_1MiB      = 1024 * 1024
 	MaxBuffers = 1
+	// udcCacheSize - Arbitrary number: exceeding this number means that in the expiry timeframe we requested pre-signed urls from
+	// more the 30 different accounts which is highly unlikely
+	udcCacheSize = 30
 
 	URLTemplate = "https://%s.blob.core.windows.net/"
 )
 
 type Adapter struct {
-	clientCache     *ClientContainerCache
+	clientCache *ClientCache
+	// udCredCache - User Delegation Credential cache used to reduce POST requests while creating pre-signed URLs
+	udCredCache     *lru.ARCCache
```

Comment:
userDeleCredCache or userCredCache just to make it more readable than 'ud' prefix.

Reply:
udcCache?